src/services/dataFetcher.ts

```
@@ -465,13 +180,15 @@ export class DataFetcher {
         reason: errorMessage,
         details: logDetails,
       });
-      DataFetcher.setTileState<TTileData>(storageKey, null, false);
+      // On error, update only lastDataRequest and lastDataRequestSuccessful, do not update data
+      const cached = storageManager.getTileState<TTileData>(storageKey);
+      const prevData =
+        cached && cached.data ? cached.data : (parser.createDefault() as unknown as TTileData);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)